src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -186,65 +206,35 @@ void icm426xxGyroInit(gyroDev_t *gyro)
     spiWriteReg(dev, ICM426XX_RA_PWR_MGMT0, ICM426XX_PWR_MGMT0_TEMP_DISABLE_OFF | ICM426XX_PWR_MGMT0_ACCEL_MODE_LN | ICM426XX_PWR_MGMT0_GYRO_MODE_LN);
     delay(15);
 
-    uint8_t odrIdx = 0;
-    bool supportedODRFound = false;
-
-    if (gyro->gyroRateKHz) {
-        const uint8_t gyroSyncDenominator = gyro->mpuDividerDrops + 1; // rebuild it in here, see gyro_sync.c
-        const uint8_t desiredODR = 8 / gyroSyncDenominator;
-        for (unsigned i = 0; i < ARRAYLEN(odrConfigLUT); i++) {
-            if (odrConfigLUT[i].odr == desiredODR) {
-                odrIdx = odrConfigLUT[i].odr_idx;
-                supportedODRFound = true;
-                break;
-            }
-        }
-    }
-
-    if (!supportedODRFound) {
-        odrIdx = 6;
+    // Get desired output data rate
+    uint8_t odrConfig;
+    const unsigned decim = llog2(gyro->mpuDividerDrops + 1);
+    if (gyro->gyroRateKHz && decim < ODR_CONFIG_COUNT) {
```

Comment:
( can you consider using ARRAYLEN, please? But both varianst are PK)
```suggestion
    if (gyro->gyroRateKHz && decim < ARRAYLEN(odrLUT)) {
```

Reply:
I was too slow... I don't mind if you change it to `ARRAYLEN()`